docs/DeveloperGuide.adoc

```diff
@@ -836,6 +836,58 @@ Use case ends.
 Use case resumes at step 2.
 
 [discrete]
+=== Use case: Undo last command
+
+*MSS*
+
+1. User requests to undo last command
+2. AddressBook confirms if user really wishes to undo last command
+3. User confirms the undo
+4. AddressBook undoes the last command
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. User cancels the undo.
++
+Use case ends.
+
+* 4a. The last command is undoable (i.e. a command that does not modify AddressBook's content)
++
+[none]
+** 4a1. AddressBook shows an error message.
++
+Use case ends.
+
+[discrete]
+=== Use case: Redo most recently undone command
+
+*MSS*
+
+1. User requests a redo of the most recently undone command
+2. AddressBook confirms if user really wishes to redo
+3. User confirms the redo
+4. AddressBook redoes the last command
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. User cancels the redo.
++
+Use case ends.
+
+* 4a. There are no `undo` commands executed previously
```

Comment:
It seems smoother if we inform the user that there are no undo commands to redo *before* we ask for confirmation? In that case, we should shift this extension to step 1.

Reply:
Since we have decided not to do any confirmation checks, and `redo` has the complementary command `undo` to reverse user action, I will presume that this is no longer necessary.